libkbfs/block_journal.go

```diff
@@ -47,20 +49,16 @@ import (
 // ID, and key_server_half, which contains the raw data for the
 // associated key server half.
 //
-// TODO: Add a mode which doesn't assume that this is the only storage
-// for TLF data, i.e. that doesn't remove files when the (known)
-// refcount drops to zero, etc.
-type bserverTlfJournal struct {
+// blockJournal is not goroutine-safe, so any code that uses it must
```

Comment:
Yes, I want a single lock to cover both the MD and bserver journals. It will eventually come into play, once we tie block operations to MD operations. We need to be able to get the latest revision from the MD journal from block ops.

I'd rather have the locking be done in `bserver_disk` for a couple of reasons: If we `double-lock` in both the block journal and the journal server, then the block journal locking may hide problems with the journal server locking, and it's somewhat confusing to have the MD journal not be locked and the block journal be locked, since they're pretty similar otherwise.